drivers/timer/nrf_rtc_timer.c

```diff
@@ -550,24 +529,27 @@ int _sys_clock_driver_init(struct device *device)
 
 u32_t _timer_cycle_get_32(void)
 {
+	u32_t ticked_cycles;
 	u32_t elapsed_cycles;
-	u32_t sys_clock_tick_count;
-	u32_t rtc_prev;
-	u32_t rtc_now;
 
-	rtc_now = RTC_COUNTER;
-	/* Discard value of RTC_COUNTER read at LFCLK transition */
-	do {
-		sys_clock_tick_count = _sys_clock_tick_count;
-		elapsed_cycles = (rtc_now - (sys_clock_tick_count *
-					     sys_clock_hw_cycles_per_tick)) &
-				 RTC_MASK;
-		rtc_prev = rtc_now;
-		rtc_now = RTC_COUNTER;
-	} while (rtc_now != rtc_prev);
-
-	return (sys_clock_tick_count * sys_clock_hw_cycles_per_tick) +
-	       elapsed_cycles;
+	/* Number of timer cycles announced as ticks so far. */
+	ticked_cycles = _sys_clock_tick_count * sys_clock_hw_cycles_per_tick;
```

Comment:
Now I realized that this information was stored in `rtc_past`, so actually there's no need to do this multiplication here.

Reply:
There is subtle difference:
```
rtc_past = (_sys_clock_tick_count * sys_clock_hw_cycles_per_tick) & RTC_MASK;
ticked_cycles = (_sys_clock_tick_count * sys_clock_hw_cycles_per_tick);
```
We are doing this multiplication because we want 32-bit timestamp which will not wrap up when RTC do so.

Reply:
However, ;-)
it seems that `rtc_past` could be stored without prior masking (then it could be used instead of the multiplication mentioned above), as in all places where it is later used the masking would be done anyway (either by using `RTC_MASK` or in hardware when `rtc_compare_set` is called).

But I guess this can be left as a field for future optimizations.